src/Features/Core/Portable/Diagnostics/DiagnosticOptions.cs

```diff
@@ -2,31 +2,31 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Collections.Immutable;
 using System.Composition;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Options.Providers;
 
-namespace Microsoft.VisualStudio.LanguageServices.Experimentation
+namespace Microsoft.CodeAnalysis.Diagnostics
 {
-    internal static class ExperimentationOptions
-    {
-        internal const string LocalRegistryPath = @"Roslyn\Internal\Experiment\";
-    }
-
     [ExportOptionProvider, Shared]
-    internal class ExperimentationOptionsProvider : IOptionProvider
+    internal sealed class DiagnosticOptions : IOptionProvider
     {
+        private const string FeatureName = "DiagnosticOptions";
+
+        public static readonly Option2<bool> LspPullDiagnosticsFeatureFlag = new(
+            FeatureName, nameof(LspPullDiagnosticsFeatureFlag), defaultValue: false,
+            new FeatureFlagStorageLocation("PullDiagnostics", prefix: "Lsp"));
```

Comment:
will have to see how this works.  not sure what the idea of hte 'prefix: Lsp' is here.